File: AD-HOC/2434/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int s = scanner.nextInt();
        int min = s;
        for (int day = 0; day < n; day++) {
            s += scanner.nextInt();
            if (s < min)
                min = s;
        }
        System.out.println(min);
        scanner.close();
    }
}

File: AD-HOC/2434/main.c
#include <stdio.h>

int main() {
    int n, s, transaction;
    scanf("%d %d", &n, &s);
    int min = s;
    for (int day = 0; day < n; day++) {
        scanf("%d", &transaction);
        s += transaction;
        if (s < min) min = s;
    }
    printf("%d\n", min);
    return 0;
}
